HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs

```diff
@@ -50,6 +55,122 @@ public ICollection<Examination> BasicSearch(BasicAppointmentSearchDTO parameters
             return actuallyAvailableAppointments;
         }
 
+        public ICollection<Examination> GetUnchangedAppointmentsForEmergency(AppointmentSearchWithPrioritiesDTO parameters)
+        {
+            parameters.InitialParameters.EarliestDateTime = GetNewStartTime(parameters.InitialParameters.EarliestDateTime);
+            parameters.InitialParameters.LatestDateTime = parameters.InitialParameters.EarliestDateTime.AddHours(2);
+
+            ICollection<Doctor> allDoctors = _doctorRepository.GetDoctorsBySpecialty(parameters.SpecialtyId);
+            List<Examination> allUnchangedAppointments = new List<Examination>();
+            foreach (Doctor doctor in allDoctors)
+            {
+                parameters.InitialParameters.DoctorJmbg = doctor.Jmbg;
+                allUnchangedAppointments.AddRange(BasicSearch(parameters.InitialParameters));
+            }
+            allUnchangedAppointments.ForEach(e => e.ExaminationStatus = ExaminationStatus.AVAILABLE);  
+
+            return allUnchangedAppointments;
+        }
```

Comment:
Firstly, we check if there is any free appointment that fits our search. If there is, it is returned to the front and immediately scheduled (not done yet).